src/main/java/sirius/biz/storage/layer3/uplink/LocalDirectoryRoot.java

```diff
@@ -0,0 +1,294 @@
+                new MutableVirtualFile(parent, fileToWrap.getName()).withExistsFlagSupplier(this::existsFlagSupplier)
+                                                                    .withDirectoryFlagSupplier(this::isDirectoryFlagSupplier)
+                                                                    .withSizeSupplier(this::sizeSupplier)
+                                                                    .withLastModifiedSupplier(this::lastModifiedSupplier)
+                                                                    .withCanDeleteHandler(this::canDeleteHandler)
+                                                                    .withDeleteHandler(this::deleteHandler)
+                                                                    .withCanCreateChildren(this::canCreateChildrenHandler)
+                                                                    .withCanCreateDirectoryHandler(this::canCreateDirectoryHandler)
+                                                                    .withCreateDirectoryHandler(this::createDirectoryHandler)
+                                                                    .withCanProvideInputStream(this::canProvideInputStreamHandler)
+                                                                    .withInputStreamSupplier(this::inputStreamSupplier)
+                                                                    .withCanProvideOutputStream(this::canCreateOutputStream)
+                                                                    .withOutputStreamSupplier(this::outputStreamSupplier)
+                                                                    .withCanRenameHandler(this::canRenameHandler)
+                                                                    .withRenameHandler(this::renameHandler)
+                                                                    .withCanMoveHandler(this::canMoveHandler)
+                                                                    .withMoveHandler(this::moveHandler)
+                                                                    .withChildren(innerChildProvider);
+        result.attach(fileToWrap);
+        return result;
+    }
+
+    private boolean moveHandler(VirtualFile file, VirtualFile newParent) {
+        try {
+            File parent = newParent.tryAs(File.class).orElse(null);
+            if (parent == null) {
+                return false;
+            }
+
+            Files.move(file.as(File.class).toPath(), new File(parent, file.name()).toPath());
+            return true;
+        } catch (Exception e) {
+            throw Exceptions.handle()
+                            .to(StorageUtils.LOG)
+                            .error(e)
+                            .withSystemErrorMessage("Layer 3/FS: Cannot move %s to %s: %s (%s)", file, name)
+                            .handle();
+        }
+    }
+
+    private boolean canMoveHandler(VirtualFile file) {
+        return !readonly && file.exists();
```

Comment:
Is this readonly flag in regards to the filesystem as a whole or the specific file? Single files could be readonly in the given filesystem too, or not?

Reply:
the idea is to mount the complete filesystem as readonly - all canXXX handlers can only try to predict if an operation is possible - the underlying FS always has the last word - sadly there is no api to ask...